scenarios/azure/aks/run.go

```diff
@@ -51,14 +53,18 @@ func Run(ctx *pulumi.Context) error {
 
 		// Deploy the agent
 		if env.AgentDeploy() {
+			// On Kata nodes, AKS uses node-name (like aks-kata-21213134-vmss000000) as only SAN in Kubelet certificate
+			// However, the DNS name aks-kata-21213134-vmss000000 is not resolvable, so cannot be used to reach Kubelet
+			// Thus we need to use `tlsVerify: false` and `and `status.hostIP` as `host` in the Helm values
 			customValues := `
 datadog:
   kubelet:
     host:
       valueFrom:
         fieldRef:
-          fieldPath: spec.nodeName
+          fieldPath: status.hostIP
     hostCAPath: /etc/kubernetes/certs/kubeletserver.crt
+	tlsVerify: false
```

Comment:
On some nodes, DNS does not resolve the nodename. The only SAN in the certificate is the node name so we can't use tlsVerify